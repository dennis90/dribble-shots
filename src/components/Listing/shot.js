import React from 'react'

import ShotFooter from './shot-footer'


const Shot = ({shot}) => {
  const getStyle = () => ({
      backgroundImage: `url(${shot.images.normal})`
  })

  return (
    <div className="shot" style={getStyle()}>
      <ShotFooter shot={shot} />
    </div>
  )
}

export default Shot
