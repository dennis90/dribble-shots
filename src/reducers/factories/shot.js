import {Factory} from 'rosie'

export const ShotFactory = new Factory()
  .sequence('id')
  .sequence('title', index => `Shot #${index}`)
  .sequence('description', index => `Shot Description #${index}`)
  .attrs({
    images: {
      hidpi: 'https://dribbble/images/hidpi',
      normal: 'https://dribbble/images/normal',
    },
    views_count: 100,
    likes_count: 100,
  })
  .attr('user', ['user'], user => UserFactory.build(user))

export const UserFactory = new Factory()
  .sequence('name', index => `User #${index}`)
  .attrs({
    avatar_url: 'https://dribbble.com/user/avatar.png',
    html_url: 'https://dribbble.com/user/',
  })
