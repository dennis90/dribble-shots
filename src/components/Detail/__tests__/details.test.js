import React from 'react'
import {shallow} from 'enzyme'

import ShotDetails from '../details'
import {ShotFactory} from '../../../reducers/factories/shot'

describe('Detail ShotDetails Component', () => {
  it('should render image, views count and likes count', () => {
    const props = ShotFactory.build({})
    const wrapper = shallow(<ShotDetails shot={props} />)
    expect(wrapper).toMatchSnapshot('Shot image, views and likes')
  })
})
