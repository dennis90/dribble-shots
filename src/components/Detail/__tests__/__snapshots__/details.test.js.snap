// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shot image, views and likes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Detail
    shot={
        Object {
            "description": "Shot Description #1",
            "id": 1,
            "images": Object {
              "hidpi": "https://dribbble/images/hidpi",
              "normal": "https://dribbble/images/normal",
            },
            "likes_count": 100,
            "title": "Shot #1",
            "user": Object {
              "avatar_url": "https://dribbble.com/user/avatar.png",
              "html_url": "https://dribbble.com/user/",
              "name": "User #1",
            },
            "views_count": 100,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="image"
>
          <img
                    alt="Shot #1"
                    src="https://dribbble/images/hidpi"
          />
</div>,
        <div
          className="info"
>
          <div
                    className="views-count"
          >
                    <span>
                              100
                               views
                    </span>
          </div>
          <div
                    className="likes-count"
          >
                    <span>
                              100
                               likes
                    </span>
          </div>
</div>,
      ],
      "className": "shot-data",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Shot #1"
            src="https://dribbble/images/hidpi"
/>,
          "className": "image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": "Shot #1",
            "src": "https://dribbble/images/hidpi",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="views-count"
>
              <span>
                            100
                             views
              </span>
</div>,
            <div
              className="likes-count"
>
              <span>
                            100
                             likes
              </span>
</div>,
          ],
          "className": "info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                100
                 views
</span>,
              "className": "views-count",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  100,
                  " views",
                ],
              },
              "ref": null,
              "rendered": Array [
                100,
                " views",
              ],
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                100
                 likes
</span>,
              "className": "likes-count",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  100,
                  " likes",
                ],
              },
              "ref": null,
              "rendered": Array [
                100,
                " likes",
              ],
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image"
>
            <img
                        alt="Shot #1"
                        src="https://dribbble/images/hidpi"
            />
</div>,
          <div
            className="info"
>
            <div
                        className="views-count"
            >
                        <span>
                                    100
                                     views
                        </span>
            </div>
            <div
                        className="likes-count"
            >
                        <span>
                                    100
                                     likes
                        </span>
            </div>
</div>,
        ],
        "className": "shot-data",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Shot #1"
              src="https://dribbble/images/hidpi"
/>,
            "className": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "alt": "Shot #1",
              "src": "https://dribbble/images/hidpi",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="views-count"
>
                <span>
                                100
                                 views
                </span>
</div>,
              <div
                className="likes-count"
>
                <span>
                                100
                                 likes
                </span>
</div>,
            ],
            "className": "info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  100
                   views
</span>,
                "className": "views-count",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    100,
                    " views",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  100,
                  " views",
                ],
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  100
                   likes
</span>,
                "className": "likes-count",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    100,
                    " likes",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  100,
                  " likes",
                ],
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
